server/fleet/scripts.go

```diff
@@ -215,14 +215,14 @@ func (hsr HostScriptResult) UserMessage(hostTimeout bool) string {
 	case -1:
 		return RunScriptScriptTimeoutErrMsg
 	case -2:
-		return "Scripts are disabled for this host. To run scripts, deploy a Fleet installer with scripts enabled."
+		return RunScriptDisabledErrMsg
 	default:
 		return ""
 	}
 }
 
 func (hsr HostScriptResult) HostTimeout(waitForResultTime time.Duration) bool {
-	return time.Now().After(hsr.CreatedAt.Add(waitForResultTime))
+	return hsr.ExitCode == nil && time.Now().After(hsr.CreatedAt.Add(waitForResultTime))
```

Comment:
This was a while ago but I believe that if the script times-out on the host, the exit code will be set to `-1`? I.e. a result _will_ be sent by fleetd - an `nil` `ExitCode` indicates that the script is pending execution.

Reply:
-1 is the exit code when the script is [killed](https://github.com/fleetdm/fleet/blob/39c55571c6cfc4184c57d88c9defdde4396c2f8b/server/fleet/scripts.go#L216-L217). There's slightly different [message](https://github.com/fleetdm/fleet/blob/39c55571c6cfc4184c57d88c9defdde4396c2f8b/server/fleet/scripts.go#L209-L210) returned if the host times out.

Reply:
Oh I see, thanks, and in this file, `HostTimeout` is only called if `ExitCode` is `nil`, but in other places it gets called without that extra check, so it makes sense to have the `ExitCode` check be part of the method. :+1: